File: 프로그래머스/unrated/181919. 콜라츠 수열 만들기/콜라츠 수열 만들기.java
class Solution {
    public int[] solution(int n) {
        
        int len = 1;
        for (int m = n; m != 1; m = m % 2 == 0 ? m / 2 : 3 * m + 1)
            len++;
        int[] answer = new int[len];
        len = 0;
        for (int m = n; m != 1; m = m % 2 == 0 ? m / 2 : 3 * m + 1)
            answer[len++] = m;
        answer[len] = 1;
        return answer;
    }
}

File: 프로그래머스/Lv.0/181919. 콜라츠 수열 만들기/콜라츠 수열 만들기.js
function solution(n) {
    var answer = [n];
    for (var x = n; x !== 1;) {
        if (x % 2 === 0)
            x /= 2;
        else
            x = x * 3 + 1;
        answer.push(x);
    }
    return answer;
}